### ISSUE TITLE
Text columns are converted aggressively to DateTime in charts

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#11069]
**Describe the bug**
Text columns are converted aggressively to DateTime in charts and x-axis shows date even when scale is set to Ordinal.

**To Reproduce**
Set viz to Bar or Line for the following query:
```
SELECT '1396-01' AS t, 40 AS c UNION ALL
SELECT '1396-02' AS t, 40 AS c UNION ALL
SELECT '1396-03' AS t, 40 AS c UNION ALL
SELECT '1396-04' AS t, 40 AS c UNION ALL
SELECT '1396-05' AS t, 40 AS c UNION ALL
SELECT '1396-06' AS t, 40 AS c UNION ALL
SELECT '1396-07' AS t, 40 AS c UNION ALL
SELECT '1396-08' AS t, 40 AS c UNION ALL
SELECT '1396-09' AS t, 40 AS c UNION ALL
SELECT '1396-10' AS t, 40 AS c UNION ALL
SELECT '1396-11' AS t, 40 AS c UNION ALL
SELECT '1396-12' AS t, 40 AS c
```

**Screenshots**
![Screenshot_00000](https://user-images.githubusercontent.com/1447303/66292161-d5bf3480-e8a0-11e9-9643-30de3f93153c.png)

Metabase 0.33.3 and `master` (commit 7b2431e2a4983a4b03dfd32072f2c884cbdd929a) with various browsers.

**Additional context**
https://discourse.metabase.com/t/text-is-considered-as-date-time-in-visualization-v0-33-3/7581

**Comment by michaeldam** [comment]
Full month results with ordinal formatted YYYY-MM, are being attributed to the first of the month as opposed to the last. When selecting ordinal, the content should not be interpreted and modified.

**Comment by paulrosenzweig** [comment]
Fixed in #11156
